### PR TITLE
aegisub: fix x86_64-linux build

### PR DESCRIPTION
**Pull request by fuzzypixelz**
###### Motivation for this change

ZHF: #122042

This fixes the Linux build for Aegisu. The main repository is clearly not active so I considered switching to the 3.3.2 update through wangqr's [fork](https://github.com/wangqr/Aegisub/releases/tag/v3.3.2), however, I wasn't too sure that was the right call (correct me if I'm wrong).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cole-h** [comment]
@ofborg eval

**Comment by austinbutler** [comment]
I get this error:

```text
building '/nix/store/2fsaxh0h82y4fibsmmxpm7m05fpr5706-1aa9215e7fc360de05da9b7ec2cd68f1940af8b2.patch.drv'...
building '/nix/store/17vldx9qmwjixmnvbhljlrbllqjjwgj5-7a6da26be6a830f4e1255091952cc0a1326a4520.patch.drv'...

trying https://github.com/Aegisub/Aegisub/commit/7a6da26be6a830f4e1255091952cc0a1326a4520.patch
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

trying https://github.com/Aegisub/Aegisub/commit/1aa9215e7fc360de05da9b7ec2cd68f1940af8b2.patch
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   655  100   655    0     0   1597      0 --:--:-- --:--:-- --:--:--  1597
hash mismatch in fixed-output derivation '/nix/store/41397xd8cd3yi0fzsapp33c6728ywx4w-7a6da26be6a830f4e1255091952cc0a1326a4520.patch':
  wanted: sha256:1my6ldrwlfxzmpdbw840adrjvzbz2yi4ixzv4abz5lcnk2c3nxf5
  got:    sha256:0k5bf5ijmpdni4xr6k073r8j27f14yz1qcp2yw6qwrb564idr97x
```

**Comment by dali99** [comment]
Switching to the wangqr fork or even AegisubDC would be the right thing to do in my opinion

File: pkgs/applications/video/aegisub/default.nix [hunk 1]
```diff
@@ -113,6 +127,10 @@ stdenv.mkDerivation rec {
     "relro"
   ];
 
+  postPatch = ''
+    sed -i 's/-Wno-c++11-narrowing/-Wno-narrowing/' {configure.ac,src/Makefile}
```
Comment (kini) [inline pkgs/applications/video/aegisub/default.nix:131]: Why not the following?

```suggestion
    sed -i 's/-Wno-c++11-narrowing/-Wno-narrowing/' configure.ac src/Makefile
```

**Comment by fuzzypixelz** [comment]
@austinbutler Thanks for your review, apparently `nix-prefetch-url` doesn't give the correct hash of the patches since I got different results running `sha256sum` on a download.

@dali99 I think that would be better for the maintainers in the short run. However, I think the fork does more than just fix the build errors (and most of these fixes are available at the master brach, another good options IMHO). So I think it would be very sneaky to fork the package from under an unsuspercting user :^). Maybe we could have a seperate package? What's more, the main repo is awaiting a new[major](https://github.com/Aegisub/Aegisub/issues/214) branch (albeit very slow moving AFAICT).

**Comment by AndersonTorres** [comment]
Indeed it was in my pile!

**Pull request merged**